astropy/coordinates/sky_coordinate.py

```diff
@@ -1163,6 +1163,40 @@ def spherical_offsets_to(self, tocoord):
         dlat = acoord.spherical.lat.view(Angle)
         return dlon, dlat
 
+    def spherical_offsets_by(self, d_lon, d_lat):
+        """
+        Computes the coordinates at the specified angular offsets.
+
+        Parameters
+        ----------
+        d_lon : angle-like
+            The angular offset in the longitude direction (e.g., RA for
+            equatorial coordinates).
+        d_lat : angle-like
+            The angular offset in the latitude direction (e.g., Dec for
+            equatorial coordinates).
+
+        Returns
+        -------
+        newcoord : `~astropy.coordinates.SkyCoord`
+            The coordinates for the location that corresponds to offsetting by
+            ``d_lat`` in the latitude direction and ``d_lon`` in the longitude
+            direction.
+
+        Notes
+        -----
+        This internally uses `~astropy.coordinates.SkyOffsetFrame` to do the
+        transformation. For a more complete set of transform offsets, use
+        `~astropy.coordinates.SkyOffsetFrame` or `~astropy.wcs.WCS` manually.
+
+        See Also
+        --------
+        spherical_offsets_to : compute the angular offsets to another coordinate
+        directional_offset_by : offset a coordinate by an angle in a direction
+        """
+        return SkyCoord(SkyOffsetFrame(d_lon, d_lat,
```

Comment:
maybe?
```suggestion
        return self.__class__(SkyOffsetFrame(d_lon, d_lat,
```

Reply:
I was going to do this, and happy to make this change, but I noticed explicit uses of `SkyCoord` elsewhere in returns, so this seems to be part of a bigger problem: `SkyCoord` isn't really subclassable #8340

Reply:
Yeah, there's also xref issues when subclassing SkyCoord.
Subclassability would be a good v5 goal

Reply:
OK I made the change, but maybe we should have a meta issue "SkyCoord is not subclass-able" or something...